### PR TITLE
webkit2gtk and gemacs rebuild

### LINKED ISSUE #5880: symbol lookup error: /usr/local/lib64/libjavascriptcoregtk-4.0.so.18: undefined symbol: u_strToUpper_68
I installed `gemacs` package, then installed [doom emacs](https://github.com/hlissner/doom-emacs). When running `gemacs` I have this error : 
```
gemacs: symbol lookup error: /usr/local/lib64/libjavascriptcoregtk-4.0.so.18: undefined symbol: u_strToUpper_68
```
I couldn't manage to reinstall gemacs with compilation (`crew reinstall -s gemacs`) due to compile errors, and also failed with `webkit2gtk_4`

### PR DESCRIPTION
**Pull request by satmandu**
Fixes #5880 

```
ldd /usr/local/lib64/libjavascriptcoregtk-4.0.so.18.18.5
        linux-vdso.so.1 (0x00007fffa65c9000)
        libglib-2.0.so.0 => /usr/local/lib64/libglib-2.0.so.0 (0x00007eb6a7027000)
        libicudata.so.69 => /usr/local/lib64/libicudata.so.69 (0x00007eb6a54cf000)
        libicui18n.so.69 => /usr/local/lib64/libicui18n.so.69 (0x00007eb6a5173000)
        libicuuc.so.69 => /usr/local/lib64/libicuuc.so.69 (0x00007eb6a4f61000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00007eb6a4f4e000)
        libgio-2.0.so.0 => /usr/local/lib64/libgio-2.0.so.0 (0x00007eb6a4d82000)
        libgobject-2.0.so.0 => /usr/local/lib64/libgobject-2.0.so.0 (0x00007eb6a4d1c000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007eb6a4cfc000)
        libz.so.1 => /usr/local/lib64/libz.so.1 (0x00007eb6a4ce0000)
        libstdc++.so.6 => /usr/local/lib64/libstdc++.so.6 (0x00007eb6a4ac3000)
        libm.so.6 => /lib64/libm.so.6 (0x00007eb6a49c0000)
        libgcc_s.so.1 => /usr/local/lib64/libgcc_s.so.1 (0x00007eb6a49a5000)
        libc.so.6 => /usr/local/lib64/libc.so.6 (0x00007eb6a47e3000)
        /usr/local/lib64/ld-linux-x86-64.so.2 (0x00007eb6a8ec3000)
        libpcre.so.1 => /usr/local/lib64/libpcre.so.1 (0x00007eb6a4759000)
        libgmodule-2.0.so.0 => /usr/local/lib64/libgmodule-2.0.so.0 (0x00007eb6a4752000)
        libmount.so.1 => /usr/local/lib64/libmount.so.1 (0x00007eb6a46ec000)
        libresolv.so.2 => /lib64/libresolv.so.2 (0x00007eb6a46d2000)
        libffi.so.7 => /usr/local/lib64/libffi.so.7 (0x00007eb6a46bf000)
        libblkid.so.1 => /usr/local/lib64/libblkid.so.1 (0x00007eb6a4660000)
        librt.so.1 => /lib64/librt.so.1 (0x00007eb6a4656000)
```

Works properly:
- [x] x86_64

Builds properly:
- [x] x86_64
- [x] i686
- [x] armv7l

### CODE REVIEW

File: packages/webkit2gtk_4.rb [hunk 1]
```diff
@@ -56,22 +57,55 @@ class Webkit2gtk_4 < Package
...
-      # system "env #{CREW_ENV_OPTIONS} \
       # Bubblewrap sandbox breaks on epiphany with
       # bwrap: Can't make symlink at /var/run: File exists
-      # ccache currently breaks gcc builds of webkit-gtk
-      system "cmake \
+      case ARCH
+      when 'x86_64'
+        system "CC=clang CXX=clang++ LD=lld NM=llvm-nm cmake \
+          -G Ninja \
+          #{CREW_CMAKE_OPTIONS.gsub('-flto', '').gsub('-ffat-lto-objects', '')} \
```
Comment (uberhacker) [inline packages/webkit2gtk_4.rb:82]: Stuff like this makes me cringe. :)

File: packages/webkit2gtk_4.rb [hunk 2]
```diff
@@ -56,22 +57,55 @@ class Webkit2gtk_4 < Package
...
+          -DENABLE_GAMEPAD=OFF \
+          -DENABLE_GLES2=ON \
+          -DENABLE_GTKDOC=OFF \
+          -DENABLE_INTROSPECTION=ON \
+          -DENABLE_MINIBROWSER=ON \
+          -DENABLE_VIDEO=ON \
+          -DENABLE_WAYLAND_TARGET=ON \
+          -DENABLE_WEB_AUDIO=OFF \
+          -DPORT=GTK \
+          -DUSE_GTK4=OFF \
```
Comment (uberhacker) [inline packages/webkit2gtk_4.rb:98]: Why OFF?

File: packages/webkit2gtk_4.rb [hunk 3]
```diff
@@ -4,27 +4,28 @@ class Webkit2gtk_4 < Package
...
-     x86_64: '34d1284e175e6ebdf078ce0ba7e08b404ebb125849470668a35162d2aa2d1daf'
+    aarch64: '64e118160d7101af91e8ffd1505d7eaf2b03a32f6a6bede1f16402d9e40e265b',
+     armv7l: '64e118160d7101af91e8ffd1505d7eaf2b03a32f6a6bede1f16402d9e40e265b',
+       i686: 'cfa921effe347074bb318eaf9cc00e2940262e2c6da452a799cf54ef9c41f12c',
+     x86_64: '45c58419c27cfa20d74f025cf6778acb79a31d7f0a7de0cf2d403f9afbd2ac9a'
   })
 
   depends_on 'atk'
   depends_on 'cairo'
+  depends_on 'ccache'
```
[the remark this answers is not in the document]
Comment (satmandu) [reply packages/webkit2gtk_4.rb:28]: yes

File: packages/webkit2gtk_4.rb (hunk 1 above)
Comment (satmandu) [reply packages/webkit2gtk_4.rb:82]: Yeah me too. Easier to set defaults and back them out when they break something though?

File: packages/webkit2gtk_4.rb (hunk 2 above)
Comment (satmandu) [reply packages/webkit2gtk_4.rb:98]: **GTK4=ON** switches the library to webkit2gtk_5.

Comment (satmandu) [reply packages/webkit2gtk_4.rb:98]: webkit2gtk_4 is explicitly built on GTK3, not GTK4.

**Pull request merged**